### ISSUE TITLE
require("dom4") in esm code

### ISSUE THREAD

**Issue by zh4ui** [palantir/blueprint#3256]
ReferenceError: Can't find variable: require in ./node_modules/@blueprintjs/core/lib/esm/index.js

where the error code is like

```
if (typeof window !== "undefined" && typeof document !== "undefined") {
    // we're in browser
    // tslint:disable-next-line:no-var-requires
    require("dom4"); // only import actual dom4 if we're in the browser (not server-compatible)
    // we'll still need dom4 types for the TypeScript to compile, these are included in package.json
}
```

#### Environment

"@blueprintjs/core": "^3.10.0",
safari on macOS Mojave

#### Expected behavior

'require' should not be in esm code

**Comment by zh4ui** [comment]
it seems a dummy "require()" can work

**Comment by giladgray** [comment]
ooh interesting yeah makes sense this would happen. don't think we can do anything about it without an API break though. upgrading to typescript 3 and using `import()` would work, but the semantics are quite different. will track solving this soon.

**Comment by adidahiya** [comment]
This was fixed by #3868, I don't think we need to change anything here at the moment.
